File: androidApp/src/main/java/at/jodlidev/esmira/androidNative/SmartphoneData.kt
package at.jodlidev.esmira.androidNative

import android.os.Build
import at.jodlidev.esmira.BuildConfig
import at.jodlidev.esmira.sharedCode.SmartphoneDataInterface
import java.util.*

/**
 * Created by JodliDev on 18.05.2020.
 */
object SmartphoneData: SmartphoneDataInterface {
	override val model: String = Build.MODEL
	override val osVersion: String = Build.VERSION.RELEASE
	override val manufacturer: String = Build.MANUFACTURER
	override val appVersion: String = BuildConfig.VERSION_NAME
	override val appType: String = "Android"
	override val lang: String get() {
		return Locale.getDefault().language
	}
}

File: iosApp/ESMira/iosNative/SmartphoneData.swift
//
// Created by JodliDev on 08.06.20.
//

import Foundation
import sharedCode

class SmartphoneData: SmartphoneDataInterface {
	var appType: String = "iOS"
	
	var appVersion: String = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
	
	var manufacturer: String = "Apple"
	
	var model: String {
		get {
			
			var systemInfo = utsname()
			uname(&systemInfo)
			let machineMirror = Mirror(reflecting: systemInfo.machine)
			
			return machineMirror.children.reduce("") { identifier, element in
				guard let value = element.value as? Int8, value != 0 else {return identifier}
				return identifier + String(UnicodeScalar(UInt8(value)))
			}
		}
	}
	
	var osVersion: String {
		get {
			let os = ProcessInfo().operatingSystemVersion
			return "\(os.majorVersion).\(os.minorVersion).\(os.patchVersion)"
		}
	}
	var lang: String {
		get {
			return Locale.current.languageCode ?? "en"
		}
	}
}
